src/test/kotlin/no/nav/syfo/brev/narmesteleder/NarmesteLederAccessServiceSpek.kt

```diff
@@ -0,0 +1,282 @@
+package no.nav.syfo.brev.narmesteleder
+
+import io.ktor.server.testing.*
+import io.mockk.clearMocks
+import io.mockk.coEvery
+import io.mockk.mockk
+import kotlinx.coroutines.runBlocking
+import no.nav.syfo.client.narmesteleder.NarmesteLederClient
+import no.nav.syfo.client.oppfolgingstilfelle.Oppfolgingstilfelle
+import no.nav.syfo.client.oppfolgingstilfelle.OppfolgingstilfelleClient
+import no.nav.syfo.dialogmote.DialogmotedeltakerService
+import no.nav.syfo.dialogmote.domain.toNarmesteLederBrevDTOList
+import no.nav.syfo.testhelper.UserConstants
+import no.nav.syfo.testhelper.generator.generateDialogmote
+import no.nav.syfo.testhelper.generator.generateDialogmotedeltakerArbeidsgiver
+import no.nav.syfo.testhelper.generator.generateDialogmotedeltakerArbeidstaker
+import no.nav.syfo.testhelper.generator.generateReferat
+import org.amshove.kluent.shouldBe
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+import java.time.LocalDate
+
+object NarmesteLederAccessServiceSpek : Spek({
+    describe(NarmesteLederAccessServiceSpek::class.java.simpleName) {
+        with(TestApplicationEngine()) {
+            start()
+
+            val anyToken = "any-token"
+            val anyCallId = "any-call-id"
+
+            val dialogmotedeltakerArbeidsgiver = generateDialogmotedeltakerArbeidsgiver()
+            val dialogmotedeltakerArbeidstaker = generateDialogmotedeltakerArbeidstaker()
+            val dialogmoteList = listOf(generateDialogmote())
+
+            val dialogmotedeltakerServiceMock = mockk<DialogmotedeltakerService>()
+            val narmesteLederClientMock = mockk<NarmesteLederClient>()
+            val oppfolgingstilfelleClientMock = mockk<OppfolgingstilfelleClient>()
+            val referatCreated6MonthsPrior = generateReferat(6L)
+            val referatCreated3MonthsPrior = generateReferat(3L)
+
+            val narmesteLederAccessService = NarmesteLederAccessService(
+                dialogmotedeltakerServiceMock,
+                narmesteLederClientMock,
+                oppfolgingstilfelleClientMock
+            )
+
+            beforeEachTest {
+                clearMocks(dialogmotedeltakerServiceMock)
+                clearMocks(narmesteLederClientMock)
+                clearMocks(oppfolgingstilfelleClientMock)
+
+                coEvery {
+                    dialogmotedeltakerServiceMock.getDialogmoteDeltakerArbeidsgiverById(
+                        any(),
+                    )
+                } returns dialogmotedeltakerArbeidsgiver
+
+                coEvery {
+                    dialogmotedeltakerServiceMock.getDialogmoteDeltakerArbeidstaker(
+                        any(),
+                    )
+                } returns dialogmotedeltakerArbeidstaker
+            }
+
+            describe("filterExpiredDocumentsInDialogmoter") {
+                it("should return brev within overlapping oppfolgingstilfelle start date") {
```

Comment:
Skjønner ikke helt hva du mener med brevet er innenfor en startdato. Tester du at brevene som returneres er sendt etter start dato? Da kan kanskje _within_  byttes ut med _sent after_?

Reply:
Ser jeg har glemt å legge til ordet `created` så det blir: `should return brev created within overlapping oppfolgingstilfelle start date`.

Er åpen for andre typer formuleringer :) kanskje: `should return brev created after overlapping oppfolgingstilfelle start date` ?

Reply:
Likte den siste. Og kan kanskje også kutte ut _overlapping_, evt si noe om hva overlapping oppfolgingstilfelle overlapper med.